frontend/src/app/main/ui/workspace/sidebar/assets.cljs

```diff
@@ -115,6 +116,53 @@
           :value (tr "labels.create")
           :on-click on-accept}]]]]]))
 
+;; ---- Common blocks ----
+
+(mf/defc asset-section
+  [{:keys [children file-id title box assets-count open?]}]
+  (let [children (if (array? children) children [children])
+        get-role #(.. % -props -role)
+        title-buttons (filter #(= (get-role %) :title-button) children)
+        content       (filter #(= (get-role %) :content) children)]
+    [:div.asset-section
+     [:div.asset-title {:class (when (not open?) "closed")}
+      [:span {:on-click (st/emitf (dwl/set-assets-box-open file-id box (not open?)))}
+       i/arrow-slide title]
+      [:span.num-assets (str "\u00A0(") assets-count ")"] ;; Unicode 00A0 is non-breaking space
+      title-buttons]
+     (when open?
+       content)]))
+
+(mf/defc asset-section-block
+  [{:keys [children role]}]
+  [:* children])
+
+(mf/defc asset-group
+  [{:keys [children group folded-groups on-change-folded-groups]}]
+  (let [path        (first group)
+        group-open? (not (contains? folded-groups path))
+
+        on-fold-group
+        (mf/use-callback
+          (mf/deps folded-groups)
+          (fn [path]
+            (fn [event]
+              (dom/stop-propagation event)
+              (on-change-folded-groups
+                (toggle-folded-group folded-groups path)))))]
+    [:*
+     (when-not (empty? path)
+       (let [[other-path last-path truncated] (cp/compact-path path 35)]
+         [:div.group-title {:class (when-not group-open? "closed")
+                            :on-click (on-fold-group path)}
```

Comment:
This is strange. Why are you passing it as argument when it's a local function that can read it from the local?
Also it's strange the use-callback here because you're always calling it on render... 
I think you can create a use-callback that depends on `path` and don't need to pass it as argument, you can read it from the scope

Reply:
This has been changed in the refactor.